src/commands/registries/azure/DockerWebhookCreateStep.ts

```diff
@@ -0,0 +1,82 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import ContainerRegistryManagementClient from 'azure-arm-containerregistry';
+import { Webhook, WebhookCreateParameters } from 'azure-arm-containerregistry/lib/models';
+import { Site } from 'azure-arm-website/lib/models';
+import { Progress } from "vscode";
+import * as vscode from "vscode";
+import { IAppServiceWizardContext, SiteClient } from "vscode-azureappservice";
+import { AzureWizardExecuteStep, createAzureClient, IActionContext } from "vscode-azureextensionui";
+import { ext } from "../../../extensionVariables";
+import { AzureRegistryTreeItem } from '../../../tree/registries/azure/AzureRegistryTreeItem';
+import { AzureRepositoryTreeItem } from '../../../tree/registries/azure/AzureRepositoryTreeItem';
+import { DockerHubRepositoryTreeItem } from '../../../tree/registries/dockerHub/DockerHubRepositoryTreeItem';
+import { RemoteTagTreeItem } from '../../../tree/registries/RemoteTagTreeItem';
+import { nonNullProp } from "../../../utils/nonNull";
+import { openExternal } from '../../../utils/openExternal';
+
+export class DockerWebhookCreateStep extends AzureWizardExecuteStep<IAppServiceWizardContext> {
+    public priority: number = 141; // execute after DockerSiteCreate
+    private _treeItem: RemoteTagTreeItem;
+    public constructor(treeItem: RemoteTagTreeItem) {
+        super();
+        this._treeItem = treeItem;
+    }
+
+    public async execute(context: IAppServiceWizardContext, progress: Progress<{
+        message?: string;
+        increment?: number;
+    }>): Promise<void> {
+        const site: Site = nonNullProp(context, 'site');
+        let siteClient = new SiteClient(site, context);
+        let appUri: string = (await siteClient.getWebAppPublishCredential()).scmUri;
+        if (this._treeItem.parent instanceof AzureRepositoryTreeItem) {
+            const creatingNewApp: string = `Creating webhook for webapp "${context.newSiteName}"...`;
+            ext.outputChannel.appendLine(creatingNewApp);
+            progress.report({ message: creatingNewApp });
+            const webhook = await this.createWebhookForApp(this._treeItem, context, appUri);
+            ext.outputChannel.appendLine(`Created webhook '${webhook.name}' with scope '${webhook.scope}', id: '${webhook.id}' and location: '${webhook.location}'`);
+        } else if (this._treeItem.parent instanceof DockerHubRepositoryTreeItem) {
+            // point to dockerhub to create a webhook
+            // http://cloud.docker.com/repository/docker/<registryName>/<repoName>/webHooks
+            const dockerhubPrompt: string = "Copy web app endpoint and browse to dockerhub";
```

Comment:
Still gotta change this button prompt.  I think how the Azure Account extension does it would be a good model.

Reply:
Thanks for pointing out. I didn't see the original comment since it was part of another thread.